modules/custom-status/custom-status.php

```diff
@@ -490,13 +510,19 @@ public function add_custom_status( $term, $args = [] ) {
 		// Reset our internal object cache
 		$this->custom_statuses_cache = [];
 
-		// Populate the inserted term with the new values, or else only the term_taxonomy_id and term_id are returned.
 		if ( is_wp_error( $inserted_term ) ) {
 			return $inserted_term;
-		} else {
-			$inserted_term = $this->get_custom_status_by( 'id', $inserted_term['term_id'] );
 		}
 
+		// Add term meta
+		if ( [] !== $required_user_ids ) {
+			update_term_meta( $inserted_term['term_id'], 'required_user_ids', $required_user_ids );
```

Comment:
`required_user_ids` passed in through `$args` will be stored in term meta here.

Reply:
Noticed that you used `update_term_meta` here instead of `add_term_update`. Was that on purpose? Figured that, that would be reserved for the `update_custom_status` method instead?

Reply:
In general the `update_*` functions in WordPress (`update_post_meta()`, `update_user_meta()`, etc) can add  meta, but they will also update an existing value if it exists. [`add_term_meta()`](https://developer.wordpress.org/reference/functions/add_term_meta/) can do the same if the `$single` parameter is set to `true`, but by default it will add a second value for a meta key if it already exists.
  
Using an `update_*` function is usually the correct choice unless you'd like non-`$single` data, like several values saved with the same key. `update_post_meta()` assumes 1:1 data, and if I saw a `add_post_meta()` I'd look closely for a 1:many relationship.